modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -486,9 +483,12 @@ private <T> CompletableFuture<T> enlistInTx(
         return postEnlist(fut, actualTx.implicit() && !singlePart, actualTx, full).handle((r, e) -> {
             if (e != null) {
                 if (actualTx.implicit()) {
+                    // TODO: IGNITE-24244
+                    long timeout = actualTx.isReadOnly() ? actualTx.timeout() : 3_000;
```

Comment:
Another constant, please

Reply:
TODO is here to change it in the next tticket